File: VB/MyXtraGrid/MyGridView.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.XtraGrid.Scrolling
Imports System.Windows.Forms
Imports System.Drawing
Imports DevExpress.Utils

Namespace MyXtraGrid
	Public Class MyGridView
		Inherits DevExpress.XtraGrid.Views.Grid.GridView
		Public Sub New()
			Me.New(Nothing)
		End Sub
		Public Sub New(ByVal grid As DevExpress.XtraGrid.GridControl)
			MyBase.New(grid)
		End Sub
		Protected Overrides ReadOnly Property ViewName() As String
			Get
				Return "MyGridView"
			End Get
		End Property

		Protected Overrides Function CreateScrollInfo() As ScrollInfo
			Dim newMyScrollInfo As ScrollInfo = MyBase.CreateScrollInfo()
			AddHandler newMyScrollInfo.VScroll.Scroll, AddressOf VScroll_Scroll
			Return newMyScrollInfo
		End Function

		Private Sub VScroll_Scroll(ByVal sender As Object, ByVal e As ScrollEventArgs)
			If e.Type = ScrollEventType.ThumbTrack Then
				Dim p As Point = GridControl.PointToScreen((CType(sender, VCrkScrollBar)).Location)
				p.Y = Control.MousePosition.Y
				MyShowHint(p, DevExpress.Utils.ToolTipLocation.TopLeft)
			End If
		End Sub

		Private Overloads Function GetToolTipText() As String
			Return String.Format("{0} - {1}", TopRowIndex, TopRowIndex + ViewInfo.RowsLoadInfo.ResultRowCount - 1)
		End Function

		Protected Overridable Sub MyShowHint(ByVal position As Point, ByVal location As ToolTipLocation)
			If GridControl Is Nothing Then
				Return
			End If
			Dim tool As ToolTipControllerShowEventArgs = GridControl.ToolTipController.CreateShowArgs()
			tool.ToolTip = GetToolTipText()
			tool.SelectedObject = Me
			tool.SelectedControl = GridControl
			tool.AutoHide = False
			tool.ToolTipLocation = location
			GridControl.ToolTipController.ShowHint(tool, position)
		End Sub
	End Class
End Namespace

File: CS/MyXtraGrid/MyGridView.cs
using System;
using DevExpress.XtraGrid.Scrolling;
using System.Windows.Forms;
using System.Drawing;
using DevExpress.Utils;

namespace MyXtraGrid {
	public class MyGridView : DevExpress.XtraGrid.Views.Grid.GridView {
		public MyGridView() : this(null) {}
		public MyGridView(DevExpress.XtraGrid.GridControl grid) : base(grid) {
		}
		protected override string ViewName { get { return "MyGridView"; } }

        protected override ScrollInfo CreateScrollInfo()
        {
            ScrollInfo newMyScrollInfo = base.CreateScrollInfo();
            newMyScrollInfo.VScroll.Scroll += VScroll_Scroll;
            return newMyScrollInfo;
        }

        void VScroll_Scroll(object sender, ScrollEventArgs e)
        {
            if (e.Type == ScrollEventType.ThumbTrack)
            {
                Point p = GridControl.PointToScreen(((VCrkScrollBar)sender).Location);
                p.Y = Control.MousePosition.Y;
                MyShowHint(p, DevExpress.Utils.ToolTipLocation.TopLeft);
            }
        }

        string GetToolTipText()
        {
            return string.Format("{0} - {1}", TopRowIndex, TopRowIndex + ViewInfo.RowsLoadInfo.ResultRowCount - 1);
        }

        protected virtual void MyShowHint(Point position, ToolTipLocation location)
        {
            if (GridControl == null) return;
            ToolTipControllerShowEventArgs tool = GridControl.ToolTipController.CreateShowArgs();
            tool.ToolTip = GetToolTipText();
            tool.SelectedObject = this;
            tool.SelectedControl = GridControl;
            tool.AutoHide = false;
            tool.ToolTipLocation = location;
            GridControl.ToolTipController.ShowHint(tool, position);
        }
	}
}